cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -389,9 +399,22 @@ public void updateApp(FullHttpRequest request, HttpResponder responder,
   @AuditPolicy(AuditDetail.REQUEST_BODY)
   public void upgradeApplication(HttpRequest request, HttpResponder responder,
                                  @PathParam("namespace-id") String namespaceId,
-                                 @PathParam("app-id") String appName) throws Exception {
-    ApplicationId appId = validateApplicationId(namespaceId, appName);
-    applicationLifecycleService.upgradeApplication(appId, createProgramTerminator());
+                                 @PathParam("app-id") String appName,
+                                 @Nullable @QueryParam("artifact-scope") String artifactScope,
```

Comment:
So the user can only pass in one scope? I believe you can actually use `Set<ArtifactScope> artifactScopes` to allow multiple scopes to be passed from the query param. Also, better not to have `-` in the query param name, as it is used as the query parameter name passed in the URI.